x-pack/plugins/apm/public/components/routing/templates/apm_service_template/analyze_data_button.tsx

```diff
@@ -51,21 +47,26 @@ export function AnalyzeDataButton() {
 
   if (
     (isRumAgentName(agentName) || isIosAgentName(agentName)) &&
-    canShowDashboard
+    rangeFrom &&
+    canShowDashboard &&
+    rangeTo
   ) {
     const href = createExploratoryViewUrl(
       {
-        'apm-series': {
-          dataType: isRumAgentName(agentName) ? 'ux' : 'mobile',
-          time: { from: rangeFrom, to: rangeTo },
-          reportType: 'kpi-over-time',
-          reportDefinitions: {
-            [SERVICE_NAME]: [serviceName],
-            ...getEnvironmentDefinition(environment),
+        reportType: 'kpi-over-time',
+        allSeries: [
+          {
+            name: `${serviceName}-response-latency`,
+            selectedMetricField: TRANSACTION_DURATION,
+            dataType: isRumAgentName(agentName) ? 'ux' : 'mobile',
+            time: { from: rangeFrom, to: rangeTo },
+            reportDefinitions: {
+              [SERVICE_NAME]: [serviceName],
+              ...(environment ? getEnvironmentDefinition(environment) : {}),
```

Comment:
Seems like this was simpler when all of the conditional logic was in the function. Now it's split between here and the body of the function.

Reply:
i agree. But typescript wasn't happy with that. Apparently switch/case isn't friendly with ts.

Case undefined: return {} wasn't resolving with ts.

it was logging warning that service.enivornment can't be assigned environemnt undefined.

Reply:
i also removed `as SeriesUrl `